src/sage/groups/matrix_gps/linear.py

```diff
@@ -307,3 +307,50 @@ def _check_matrix(self, x, *args):
         else:
             if x.determinant() == 0:
                 raise TypeError('matrix must non-zero determinant')
+
+    def order(self):
+        """
+        Return the order of ``self``.
+
+        EXAMPLES::
+
+            sage: G = SL(3, GF(5))
+            sage: G.order()
+            372000
+
+        TESTS:
+
+        Check if :trac:`36876` is fixed::
+            sage: SL(1, QQ).order()
+            1
+            sage: SL(2, ZZ).cardinality()
+            +Infinity
+
+        Check if :trac:`35490` is fixed::
+            sage: q = 7
+            sage: FqT.<T> = GF(q)[]
+            sage: N = T^2+1
+            sage: FqTN = QuotientRing(FqT, N*FqT)
+            sage: S = SL(2, FqTN)
+            sage: S.is_finite()
+            True
+            sage: S.order()
+            117600
+        """
+        from sage.rings.infinity import Infinity
+        from sage.all import prod
```

Comment:
This should be `from sage.misc.misc_c import prod`, and it might be better (if it works) to move both import statements to line 69 or so.